airbyte-server/src/main/java/io/airbyte/server/handlers/MigrationHandler.java

```diff
@@ -0,0 +1,216 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.server.handlers;
+
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigPersistence;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.DefaultConfigPersistence;
+import io.airbyte.config.persistence.PersistenceConstants;
+import io.airbyte.scheduler.persistence.JobPersistence;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.zip.GZIPOutputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MigrationHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MigrationHandler.class);
+  private static final String ARCHIVE_FILE_NAME = "airbyte_config_data";
+  private static final String VERSION_FILE_NAME = "VERSION";
+
+  private final ConfigRepository configRepository;
+  private final JobPersistence jobPersistence;
+
+  public MigrationHandler(final ConfigRepository configRepository, final JobPersistence jobPersistence) {
+    this.configRepository = configRepository;
+    this.jobPersistence = jobPersistence;
+  }
+
+  public Path exportData() {
+    try {
+      // Create temp folder where to export data
+      final Path tempFolder = Files.createTempDirectory("airbyte_archive");
+      FileUtils.forceDeleteOnExit(tempFolder.toFile());
+      exportAirbyteConfig(tempFolder);
+      exportAirbyteDatabase(tempFolder);
+      exportVersionFile(tempFolder);
+      final Path archive = createArchive(tempFolder);
+      FileUtils.deleteDirectory(tempFolder.toFile());
+      return archive;
+    } catch (IOException e) {
+      LOGGER.error("Export Data failed.");
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void exportAirbyteConfig(Path tempFolder) {
+    LOGGER.info(String.format("Exporting Airbyte Configs to %s", tempFolder));
+
+    // TODO Change persistence to YAML persistence instead of Default
+    final ConfigPersistence persistence = new DefaultConfigPersistence(tempFolder, configRepository.getAirbyteVersion());
+    final ConfigRepository tmpConfigRepository = new ConfigRepository(persistence);
```

Comment:
i'm wondering if we need this tmpConfigRepository at all. what if we added to the `ConfigPersistence` interface an `exportData` method? this method would return a path to the data on file system. internally this method would just copy the entire config repository directory? does this make life way easier?

Reply:
maybe something like this (roughly, i haven't handled exceptions and i always use the wrong copy function) but you'll get the idea:
```
  @Override
  public Path exportData() {
    final Path configPersistenceExportPath = Files.createTempDirectory(Path.of("/tmp"), "config_persistence_export");
    synchronized (lock) {
      return FileUtils.copyDirectory(storageRoot.toFile(), configPersistenceExportPath.toFile());
    }
  }
```

Reply:
I think I still need to do that since I want to change how they are persisted (change the format of the file)?

Reply:
Besides I will have to persist AIrbyte Database as files too, so it does make it easier to create the tarball afterward to have everything properly laid out in the same temp folder I think

Reply:
I think we are talking about different things. I'm just saying you copy the config directory. Not suggesting you tarball it until you do the other cleaning and package it with the db.

Reply:
Thanks for your help!! 

I pushed new code that actually do a different persistence in the temp file now